be/src/olap/rowset/segment_v2/page_compression.cpp

```diff
@@ -88,5 +88,33 @@ Status PageCompressor::compress(const std::vector<Slice>& raw_data,
     return Status::OK();
 }
 
+Status PageCompressor::compress(const std::vector<Slice>& raw_data,
+                                OwnedSlice* compressed_data, bool* compressed) {
+    size_t uncompressed_bytes = Slice::compute_total_size(raw_data);
+    size_t max_compressed_bytes = _codec->max_compressed_len(uncompressed_bytes);
+    _buf.resize(max_compressed_bytes + 4);
+    Slice compressed_slice(_buf.data(), max_compressed_bytes);
```

Comment:
compressed_slice and compressed_data looks alike. Better to choose another name.

Reply:
change it to compression_buffer